xds/internal/xdsclient/callback_serializer.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xdsclient
+
+import (
+	"google.golang.org/grpc/internal/buffer"
+	"google.golang.org/grpc/internal/grpcsync"
+)
+
+// CallbackSerializer provides a mechanism to schedule callbacks in a
+// synchronized manner. It provides a FIFO guarantee on the order of execution
+// of scheduled callbacks.
+//
+// New callbacks can be scheduled by invoking the Schedule() method. To cleanup
+// resources used by the serializer and to discard any queued callbacks, the
+// Close() method needs to be called.
+//
+// This type is safe for concurrent access.
+type CallbackSerializer struct {
```

Comment:
Why is this exported?  Will other packages be using it?  If so, move it to somewhere else?  Maybe even `grpcsync` itself?

Reply:
I initially had a comment with a TODO saying move this to somewhere outside of the xds tree so that other packages will have access to it. But I can't think of anyone who needs it now. So, I will unexport it now, and we can always move it to some place appropriate when the time comes. Thanks.